runtime/tests/Test_LightningMeasures.cpp

```diff
@@ -771,6 +999,38 @@ TEMPLATE_LIST_TEST_CASE("Var(Tensor(Hamiltonian(NamedObs[]), NamedObs)) test", "
     CHECK(sim->Var(thz) == Approx(0.64).margin(1e-5));
 }
 
+// TEMPLATE_LIST_TEST_CASE("Var(Tensor(Hamiltonian(NamedObs[]), NamedObs)) shots test",
+// "[Measures]",
+//                         SimTypes)
+// {
+//     std::unique_ptr<TestType> sim = std::make_unique<TestType>();
+
+//     // state-vector with #qubits = n
+//     constexpr size_t n = 4;
+//     std::vector<QubitIdType> Qs;
+//     Qs.reserve(n);
+//     for (size_t i = 0; i < n; i++) {
+//         Qs.push_back(sim->AllocateQubit());
+//     }
+
+//     constexpr size_t num_shots = 5000;
+//     sim->SetDeviceShots(num_shots);
+
+//     sim->NamedOperation("PauliX", {}, {Qs[0]}, false);
+//     sim->NamedOperation("PauliY", {}, {Qs[1]}, false);
+//     sim->NamedOperation("Hadamard", {}, {Qs[2]}, false);
+//     sim->NamedOperation("PauliZ", {}, {Qs[3]}, false);
+
+//     ObsIdType px = sim->Observable(ObsId::PauliX, {}, {Qs[2]});
+//     ObsIdType py = sim->Observable(ObsId::PauliY, {}, {Qs[1]});
+//     ObsIdType pz = sim->Observable(ObsId::PauliZ, {}, {Qs[0]});
+//     ObsIdType hxy = sim->HamiltonianObservable({0.4, 0.8}, {px, py});
+//     ObsIdType thz = sim->TensorObservable({hxy, pz});
+
+//     CHECK(sim->Var(thz) == Approx(0.64).margin(5e-2)); // TODO: Lightning failed!,
```

Comment:
@multiphaseCFD this is related to the Lightning bug that we discussed. We should uncomment this after merging the fix.